src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs

```diff
@@ -0,0 +1,93 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.Models;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.ProviderModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRMRecoveryServicesRecoveryPoint"), OutputType(typeof(List<AzureRmRecoveryServicesRecoveryPointBase>), typeof(AzureRmRecoveryServicesRecoveryPointBase))]
+    public class GetAzureRMRecoveryServicesRecoveryPoint : RecoveryServicesBackupCmdletBase
+    {
+        internal const string DateTimeFilterParameterSet = "DateTimeFilter";
+        internal const string RecoveryPointIdParameterSet = "RecoveryPointId";
+
+        [Parameter(Mandatory = true, ParameterSetName = DateTimeFilterParameterSet, HelpMessage = "", ValueFromPipeline = false)]        
+        [ValidateNotNullOrEmpty]
+        public DateTime StartDate { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = DateTimeFilterParameterSet, HelpMessage = "", ValueFromPipeline = false)]        
+        [ValidateNotNullOrEmpty]
+        public DateTime EndDate { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = DateTimeFilterParameterSet, HelpMessage = "", ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, ParameterSetName = RecoveryPointIdParameterSet, HelpMessage = "", ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public AzureRmRecoveryServicesItemBase Item { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = RecoveryPointIdParameterSet, HelpMessage = "", ValueFromPipeline = false)]
+        [ValidateNotNullOrEmpty]
+        public string RecoveryPointId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            ExecutionBlock(() =>
+            {
+                //Validate start time < end time
+                base.ExecuteCmdlet();            
+
+                Dictionary<System.Enum, object> parameter = new Dictionary<System.Enum, object>();
+                parameter.Add(GetRecoveryPointParams.Item, Item);
+
+                if(this.ParameterSetName == DateTimeFilterParameterSet)
+                {
+                    //User want list of RPs between given time range
+                    if (StartDate >= EndDate)
+                    {
+                        throw new Exception("End date should be greated than start date"); //tbd: Correct nsg and exception type
```

Comment:
I added classes for different types of messages we have to print. Please add this as a variable in the same. It'll be easier later to give these to PM and get them scrubbed.